src/components/Search/index.tsx

```diff
@@ -105,20 +111,41 @@ const handleSearchResult = async (
 const Search: FC<{
   content?: string
   hasButton?: boolean
+  truncateTypeHash?: boolean
   onEditEnd?: () => void
-}> = memo(({ content, hasButton, onEditEnd }) => {
+}> = memo(({ content, hasButton, onEditEnd, truncateTypeHash }) => {
   const isMobile = useIsMobile()
   const { t } = useTranslation()
+  const searchByType = cacheService.get(SEARCH_TYPE_KEY) || 'id'
+  const [isSearchByNames, setIsSearchByNames] = useState(searchByType === 'name')
+  const [searchByNameResults, setSearchByNameResults] = useState<UdtQueryResult[] | null>(null)
   const history = useHistory()
-  const SearchPlaceholder = t('navbar.search_placeholder')
   const [searchValue, setSearchValue] = useState(content || '')
-  const [placeholder, setPlaceholder] = useState(SearchPlaceholder)
+  const placeholder = isSearchByNames ? t('navbar.search_by_name_placeholder') : t('navbar.search_placeholder')
```

Comment:
It seems that it can be directly moved into the tsx code.

Reply:
inlining is OK, while extracting this viarable makes it more clean and simple on tsx code

Reply:
I believe this is actually a decrease in code readability because humans read code sequentially, and any jumps or navigations decrease readability unless they provide significant benefits or improved readability.

Since there is no code reuse involved here, I think keeping it in the .tsx file is a better choice for readability.